plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -61,7 +64,7 @@ struct SoftwareModuleUpdate {
     pub path: Option<String>,
 }
 
-fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
+fn run(operation: PluginOp, module_type: &str) -> Result<ExitStatus, InternalError> {
```

Comment:
Doesn't the `update-list` command need any updates to support safe tedge updates? I'd have expected the `update-list` for `tedge` plugin to at least stop all the tedge services before individual packages are updated? Otherwise I assume that `apt` will stop the services for a package (by running its "pre-upgrade" maintainer scripts) before its upgraded and start the upgraded version back again, package after package (eg. `tedge-agent` is upgraded first, followed by `tedge-mapper`). Unless `apt` is smart enough to run the "pre-upgrade" maintainer scripts of all packages passed to it before updating those packages.

If the services are stopped and restarted on package level, we'll have to deal with additional complexities of version compatibility between different components, like newer version of one package(say `tedge-mapper`) running while an older version of a different package (say, `tedge-agent`) is running. There could be incompatibilities between these versions in terms of messages they exchange. So wouldn't it be safer if we stop everything before the updates are applied, rather than letting apt stop and start individual services at a package level?

Reply:
You are right. That's a point I am currently about to add to the plugin. My approach is as below:

Before any upgrade happens identifying all configured bridges, and doing `tedge disconnect` for each.

After processing the entire update doing `tedge connect` for all bridges that were configured before.